### PR TITLE
Add GHC 9.2 support

### PR DESCRIPTION
**Pull request by srid**
See #35 for other PRs

### CODE REVIEW

File: stack-9.2.1.yaml.lock [hunk 1]
```diff
@@ -0,0 +1,706 @@
+# This file was autogenerated by Stack.
```
Comment (NorfairKing) [inline stack-9.2.1.yaml.lock:1]: Nit: This file shouldn't be comitted.

**Comment by srid** [comment]
Sorry I messed up my Git branches, and this now includes #37. Let me know if I should remove that.

**Comment by srid** [comment]
@sydtest How do I test this for 8.10 in nix-shell? Currently: `stack build` in nix-shell is unable to find the compiler:

```
❯ stack build
WARNING: nix-pre-commit-hooks: git command not found; skipping installation.
I don't know how to install GHC on your system configuration, please install manually
```

Aside from that (and consolidation of overrides in stack yaml post merge?) this PR should be ready for merge.

**Comment by NorfairKing** [comment]
`nix-build ci.nix` should do it.

**Comment by NorfairKing** [comment]
Done as part of #48

**Pull request closed without merging**